galsim/chromatic.py

```diff
@@ -227,6 +460,152 @@ def drawImage(self, bandpass, image=None, integrator=None, **kwargs):
+        """
+        # Check whether the object has had some chromatic transformation applied to it.  Since we
+        # are interpolating between stored images, this is not a situation that _interp_drawImage
+        # can handle.  We have to lose the interpolation.
+        if hasattr(self, '_A') and self._chromaticTransformation(bandpass):
+            import warnings
+            warnings.warn("Cannot render image with chromatic transformation applied to it"
+                          " using interpolation between stored images.  Reverting to "
+                          "non-interpolated version.")
+            self.noInterpolation()
+            return self._normal_drawImage(bandpass, image=image,
+                                          integrator=integrator, **kwargs)
+
+        # Note that integrator is not actually selectable.
+        if integrator is not None:
+            import warnings
+            warnings.warn("Cannot choose image integrator when using stored images! "
+                          "Ignoring this keyword argument.")
+
+        # setup output image (semi-arbitrarily using the bandpass effective wavelength).
+        # Note: we cannot just use self._imageAtWavelength, because that routine returns an image
+        # with whatever pixel scale was required to sample all the images properly.  We want to set
+        # up an output image that has the requested pixel scale, which might change the image size
+        # and so on.
+        prof0 = self._evaluateAtWavelength(bandpass.effective_wavelength)
+        image = prof0.drawImage(image=image, setup_only=True, **kwargs)
+        # Remove from kwargs anything that is only used for setting up image:
+        kwargs.pop('dtype', None)
+        kwargs.pop('scale', None)
+        kwargs.pop('wcs', None)
+        kwargs.pop('nx', None)
+        kwargs.pop('ny', None)
+
+        # determine combined self.wave_list and bandpass.wave_list
+        wave_list = self._getCombinedWaveList(bandpass)
+
+        # merge self.wave_list into bandpass.wave_list since we are using a sampling integrator
+        bandpass = galsim.Bandpass(galsim.LookupTable(wave_list, bandpass(wave_list),
+                                                      interpolant='linear'))
+
+        # The integration is carried out using the following two basic approaches:
+        # (1) We use linear interpolation between the stored images to get an image at a given
+        #     wavelength.
+        # (2) We use the midpoint rule for integration.
+
+        # Figure out the dwave for each of the wavelengths in the bandpass.
+        dw = [bandpass.wave_list[1]-bandpass.wave_list[0]]
+        dw.extend(0.5*(bandpass.wave_list[2:]-bandpass.wave_list[0:-2]))
+        dw.append(bandpass.wave_list[-1]-bandpass.wave_list[-2])
+        # Set up arrays to accumulate the weights for each of the stored images.
+        weight_fac = np.zeros(len(self.waves))
+        for idx, w in enumerate(bandpass.wave_list):
```

Comment:
I don't think this will work very well if the band pass is a simple rectangle approximation.  In the normal drawImage, Josh does a union of the bandpass wave_list and the object's wave_list.  I think you want to do the same here with self.waves.